scheduler/scheduler-server/src/main/java/org/ow2/proactive/utils/JettyStarter.java

```diff
@@ -377,6 +399,84 @@ private WebAppContext createWebAppContext(String contextPath, String[] virtualHo
+        logger.debug(IAMConfiguration.IAM_URL + ": " + System.getProperty(IAMConfiguration.IAM_URL));
+        logger.debug(IAMConfiguration.IAM_LOGIN + ": " + System.getProperty(IAMConfiguration.IAM_LOGIN));
+        logger.debug(IAMConfiguration.PA_SERVER_NAME + ": " + System.getProperty(IAMConfiguration.PA_SERVER_NAME));
+    }
+
+    /**
+     * Add IAM authentication and authorization filters to PA microservices webapp contexts
+     */
+    private void addIAMFilters(WebAppContext webApp) {
+
+        String context = webApp.getContextPath();
+
+        if ((context.equals("/studio") || context.equals("/scheduler") || context.equals("/rm") ||
+             context.equals("/automation-dashboard"))) {
+
+            // set a jetty context-param to remove jessionid from URLs
+            webApp.setInitParameter("org.eclipse.jetty.servlet.SessionIdPathParameterName", "none");
+
+            // set context-param to acquire filters' configurations from system properties
+            webApp.setInitParameter("configurationStrategy", "SYSTEM_PROPERTIES");
+
+            // Add SingleSignOutHttpSessionListener
+            webApp.addEventListener(new org.jasig.cas.client.session.SingleSignOutHttpSessionListener());
+
+            // Add CAS Single Sign Out Filter
+            webApp.addFilter(org.jasig.cas.client.session.SingleSignOutFilter.class,
+                             "/*",
+                             EnumSet.of(DispatcherType.REQUEST,
+                                        DispatcherType.INCLUDE,
+                                        DispatcherType.FORWARD,
+                                        DispatcherType.ASYNC,
+                                        DispatcherType.ERROR));
+
+            // Add CAS Authentication Filter
+            webApp.addFilter(org.jasig.cas.client.authentication.AuthenticationFilter.class,
+                             "/*",
+                             EnumSet.of(DispatcherType.REQUEST,
+                                        DispatcherType.INCLUDE,
+                                        DispatcherType.FORWARD,
+                                        DispatcherType.ASYNC,
+                                        DispatcherType.ERROR));
+
+            // Add CAS Validation Filter
+            webApp.addFilter(org.jasig.cas.client.validation.Cas30ProxyReceivingTicketValidationFilter.class,
+                             "/*",
+                             EnumSet.of(DispatcherType.REQUEST,
+                                        DispatcherType.INCLUDE,
+                                        DispatcherType.FORWARD,
+                                        DispatcherType.ASYNC,
+                                        DispatcherType.ERROR));
+
+            // Add CAS HttpServletRequest Wrapper Filter
+            webApp.addFilter(org.jasig.cas.client.util.HttpServletRequestWrapperFilter.class,
+                             "/*",
+                             EnumSet.of(DispatcherType.REQUEST,
```

Comment:
These EnumSet repeat 4 times. Maybe you can re-use it

Reply:
done. Thanks